task.sql

```
@@ -9,9 +9,34 @@ CREATE TABLE Countries (
 ) ENGINE=InnoDB;
 
 -- Create a table for caching GeoIP data (Columns: ID, IP Range, CountryID)
+CREATE TABLE GeoIPCache (
+    ID INT,
+    IPRange INT,
+    CountryID INT,
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CountryID) REFERENCES Countries(ID)
```

Good choice using the `MEMORY` engine for a cache table. However, it's important to know that the `MEMORY` storage engine does not support `FOREIGN KEY` constraints. While your syntax is correct and won't cause an error, MySQL will silently ignore this constraint and won't enforce referential integrity.